### PR TITLE
Enhanced release workflow to update self-hosted pack index.

### PR DESCRIPTION
**Pull request by JonatanAntoni**
- Add GitHub pack-index branch as pack URL
- Add GitHub repository to PDSC file
- Enhance gen_changelog to put specific download URLs to each release in pdsc format
- Enhance workflow to update pack-index branch

### CODE REVIEW

File: Scripts/git/gen_pack_index.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+cat <<EOF
+<?xml version="1.0" encoding="UTF-8" ?>
+<index schemaVersion="1.0.0" xs:noNamespaceSchemaLocation="PackIndex.xsd" xmlns:xs="http://www.w3.org/2001/XMLSchema-instance">
+  <vendor>ARM</vendor>
+  <url>https://raw.githubusercontent.com/ARM-software/CMSIS-DSP/pack-index/</url>
```
Comment (christophe0606) [inline Scripts/git/gen_pack_index.sh:7]: When I try this URL in a browser, I am getting:

400: Invalid request

I don't know what's this URL is useful for in the pdsc ?

File: Scripts/git/gen_pack_index.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  <timestamp>$(date -u +%FT%TZ)</timestamp>
+  <pindex>
+    <pdsc url="https://raw.githubusercontent.com/ARM-software/CMSIS-DSP/pack-index/" vendor="ARM" name="CMSIS-DSP" version="$1"/>
```
Comment (christophe0606) [inline Scripts/git/gen_pack_index.sh:10]: Same problem with this URL. I am getting a 400: Invalid request

**Comment by christophe0606** [comment]
If I understand well, the system will never try to access above URLs as a folder. They will always be accessed with a filename concatenated to the URL like with:

https://raw.githubusercontent.com/ARM-software/CMSIS-DSP/pack-index/CMSIS-DSP.pidx

Right ?

**Comment by jkrech** [comment]
This is documented here: https://open-cmsis-pack.github.io/Open-CMSIS-Pack-Spec/main/html/createPackPublish.html

This url is used to download pack versions. In the most straight forward use case a folder is hosted on the web, which contains:
a) the pdsc file of the latest pack version
b) all pack versions

Note: all packs from vendor ARM/Keil are today hosted here: https://www.keil.com/pack
Therefore use: `<url>https://www.keil.com/pack/</url>`

**Review by christophe0606 — APPROVED** [review]

**Comment by jkrech** [comment]
@christophe0606, yes the system always accesses specific files.
I need to discuss with @JonatanAntoni. I currently don't see a need for generating a package index file.

**Comment by JonatanAntoni** [comment]
@jkrech, if we don't need the separate index would be fine. Do we just need to add the pdsc url into the Arm/Keil index and it will pick new versions up, automatically?

**Pull request merged**